src/Auth/BasicAuthenticate.php

```diff
@@ -75,6 +77,11 @@ public function getUser(ServerRequest $request)
     {
         $username = $request->env('PHP_AUTH_USER');
         $pass = $request->env('PHP_AUTH_PW');
+        $encoding = Configure::read('App.encoding');
+        if ($encoding && in_array($encoding, mb_list_encodings())) {
+            $username = mb_convert_encoding($username, $encoding, self::ENCODING_SOURCES);
```

Comment:
Late static binding (static::) helps to allow extensions to work with their own const.